test/benchmark/BenchmarkIngest_test.go

```diff
@@ -112,6 +112,47 @@ func BenchmarkIngest(b *testing.B) {
 	b.Logf("\nTest complete. %d samples added to %s\n", count, tsdbPath)
 
 	// TODO: Make sure all samples have been ingested properly, i.e. execute `count` for all metrics and compare with expected
+	if err := validateCountOfSamples(adapter, count, testStartTimeMs, testEndTimeMs); err != nil {
+		b.Error(err)
+	}
+}
+
+func validateCountOfSamples(adapter *tsdb.V3ioAdapter, expected int, startTimeMs, endTimeMs int64) error {
```

Comment:
maybe pass interface `TB` (mutual between T and B tests) so that the valudation will fail the test directly instead of checking it on every test